tests/daemon/test-machine-id-provider.c

```diff
@@ -153,11 +153,20 @@ test_machine_id_provider_can_get_tracking_id (MachineIdTestFixture *fixture,
   g_autoptr(EmerMachineIdProvider) id_provider =
     emer_machine_id_provider_new_full (fixture->tracking_id_file_path);
   uuid_t id;
-  g_assert (emer_machine_id_provider_get_id (id_provider, id));
+
+  // simultaneous check if we can get a unparsed_id directly to show in UI
+  g_autofree gchar *unparsed_id_direct = NULL;
+
+  g_assert (emer_machine_id_provider_get_id (id_provider, &unparsed_id_direct, id));
   gchar unparsed_id[HYPHENS_IN_ID + FILE_LENGTH];
   uuid_unparse_lower (id, unparsed_id);
   g_autofree gchar* unhypenated_id = unhyphenate_uuid (unparsed_id);
   g_assert_cmpstr (TESTING_TRACKING_ID, ==, unhypenated_id);
+
+  // adjust \n to match  TESTING_TRACKING_ID
```

Comment:
Can we knock the `\n` out of the `TESTING_TRACKING_ID` constant instead? This code is not safe because it's not guaranteed that unparsed_id_direct is at least 34 elements long.

Reply:
The reasoning present the test file doesn't apply 100% here (because this is different from /etc/machine-id) but I think we should still respect it?
```
 * The expected size in bytes of the file located at
 * #EmerMachineIdProvider:path.
 * According to http://www.freedesktop.org/software/systemd/man/machine-id.html
 * the file should be 32 lower-case hexadecimal characters followed by a
 * newline character.
```

Reply:
My point is that it is not guaranteed that the `unparsed_id_direct` buffer returned is 34 bytes long (though in practice it will be the case), so it's not safe to modify it on the lines below. Another option would be to use:

```c
g_assert_true (strlen (unparsed_id_direct) == 32);
g_assert_cmpmem (TESTING_TRACKING_ID, 32, unparsed_id_direct, 32);
```

or similar.

Reply:
What we are trying to test here is that the value returned by `emer_machine_id_provider_get_id()` is what we expect: the 32 hexits from `TESTING_TRACKING_ID`, with no trailing newline, because that's what we want to expose over D-Bus.

Reply:
> Another option would be to use:
> 
> ```c
> g_assert_true (strlen (unparsed_id_direct) == 32);
> g_assert_cmpmem (TESTING_TRACKING_ID, 32, unparsed_
> ```

Adopted this approach for now. Addressed, please review. @wjt

Reply:
Looks good though I took the liberty of tweaking it slightly. Will merge once CI passes.